Mithril update 2024-06-05 (#407)
IntersectMBO/cardano-updates: blog/2024-06-05-mithril.md

---
title: Mithril Team Update
slug: 2024-06-05-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> This week, the Mithril team continued implementing the certification of Cardano transactions in Mithril networks by improving the throughput of the prover route of the aggregator with the usage of resource pooling and making progress on low-latency certification by completing the transition to a chain point based beacon used to sign the transactions at a faster pace. They also kept investigating prover performance bottlenecks, worked on fixing some bug that allowed some transactions not already signed to appear as such, and kept implementing the import of transactions with the ChainSync mini-protocol and Pallas.
> 
> Additionally, the team started preparing the threat modeling page on the documentation website and kept investigating some error messages of the Cardano node which could be due to the way the Pallas client handles connections to mini-protocols.

Reviewer: ```suggestion
This week, the Mithril team continued implementing the certification of Cardano transactions in Mithril networks. They improved the throughput of the prover route of the aggregator by using resource pooling and made progress on low-latency certification by transitioning to a chain point-based beacon, which allows for faster transaction signing. They also kept investigating prover performance bottlenecks, worked on fixing a bug that caused unsigned transactions to appear as signed, and continued implementing the import of transactions with the ChainSync mini-protocol and Pallas.

Additionally, the team started preparing the threat modeling explainer for the documentation website and kept investigating some Cardano node error messages that could be caused by how the Pallas client handles connections to mini-protocols.
```
